tensorboard/webapp/app_routing/effects/app_routing_effects.ts

```diff
@@ -410,23 +411,22 @@ export class AppRoutingEffects {
           })
         );
       }),
-      withLatestFrom(this.store.select(getKnownNamespaceIds)),
-      tap(([{routeMatch, options}, knownNamespaceIds]) => {
+      withLatestFrom(this.store.select(getRehydratedDeepLinks)),
+      tap(([{routeMatch, options}, rehydratedDeepLinks]) => {
         // Possibly rehydrate state from the URL.
-        //
-        // We do this on "browser initiated" events (like a page load or
-        // navigation in browser history) but we only do this when we don't yet
-        // have in-memory state for the namespace being navigated to.
 
-        const isKnownNamespace =
-          options.namespaceUpdate.option ===
-            NamespaceUpdateOption.FROM_HISTORY &&
-          knownNamespaceIds.has(options.namespaceUpdate.namespaceId);
+        if (!options.browserInitiated || !routeMatch.deepLinkProvider) {
+          return;
+        }
 
         if (
```

Comment:
Can we have some comment explaining here? For example, the params in this namespace has been rehydrated once, so we don't do anything here.

Reply:
Done.